### DISCUSSION TITLE
BEAR WATCHER - bearForm not showing upon load

### DISCUSSION THREAD

**Discussion by cmeffley** [nss-evening-cohort-14/evening-client-side#121] [Help Me!] [answered]
Chris Meffley (another help ticket, oye vey)

Other Tickets?
#106 - issue was importing from wrong module, not the issue here
#60 - issue was trying to make a loop instead of function. I have a function for this issues

Trying to get the bear form to show up when the page is loaded. It's the same thing that we did for the modular practice, but for whatever reason, I am unable to get the form to render on the page. I've had similar issues, but when comparing things to what I've done in the past, I don't see a difference / where I've gone wrong.

No error messages

Links: 
https://css-tricks.com/crafting-reusable-html-templates/
https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement
https://codereview.stackexchange.com/questions/10451/how-to-add-forms-dynamically-in-javascript#:~:text=Example%20speaks%20for%20it%20self,1px%22%3E')%3B%20document.

What have I tried:
- The function wasn't working in bearForm.js so moved it to main.js so there wasn't any export/import issues. Still doesn't work.
- Put the form in the HTML and it showed up normally upon page load.
- console.warn(divId) - got body, which is the correct div that the form needs to render in
- console.warn(selectedDiv) - got `<body></body>` which is also where the form needs to render
- console.warn(screen) - the form shows up correctly
```
const printToDom = (divId, textToPrint) => {
  console.warn(divId);
  const selectedDiv = document.querySelector(divId);
  console.warn(selectedDiv);
  selectedDiv.innerHtml = textToPrint;
};

const bearForm = () => {
  const screen = `<div class="jumbotron">
  <h1 class="display-4">Welcome to Bear Watcher!</h1>
  <h2>How fat can they get?</h2>
  <p class="lead">This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.</p>
  <hr class="my-4">
  <p>It uses utility classes for typography and spacing to space content out within the larger container.</p>
  <form class="text-center" id="form">
    <div class="mb-3">
    <label for="addNewBear" class="form-label">Add New Bear</label>
    <input type="text" class="form-control" id="addBear" required>
    <label for="addBearPicture" class="form-label">Add Picture</label>
    <input type="text" class="form-control" id="bearPicture" required>
    </div>
    <button type="submit" class="btn btn-primary" id="add1">Add Bear</button>
    </form>
</div>`;
  console.warn(screen);
  printToDom('body', screen);
};

const init = () => {
  bearForm();
};

init();
```

Repo links:
Pull Request with changes: https://github.com/nss-nightclass-projects/ASSIGNMENT-Bear-Watcher/compare/main...cmeffley:bearform?expand=1
Bear Watcher Repo: https://github.com/cmeffley/ASSIGNMENT-Bear-Watcher

**Comment by ajawashington** [comment, marked as answer]
EDITED: https://github.com/cmeffley/ASSIGNMENT-Bear-Watcher/blob/466f406a6f87c0ea154ae1a3ddf1344262b043e5/src/javascripts/main.js#L9 

Case check on this line. ✨

**Reply by cmeffley** [reply]
It worked!! Finally! My last two issues have been small typing errors (but of course I've looked at all other possibilities), is there a good way to prevent that or now that I know, just be more diligent and looking at every little detail?

**Reply by TrinityChristiana** [reply]
Really just clearing those errors like a game of whack a mole until it starts to work. That's the beauty of JavaScript. It will usually tell you when something's wrong (sometimes what's wrong). So things like console logging like we showed you could really help.
